ui/lib/homepage.dart

```diff
@@ -46,45 +70,81 @@ class Homepage extends StatelessWidget {
             ),
           ),
           child: Center(
-            child:
-            Container(
-              margin: const EdgeInsets.fromLTRB(0, 30, 0, 0),
-              constraints: const BoxConstraints(maxWidth: 1100),
-              alignment: Alignment.center,
-              child: Column(
-                children: [
-                  FeaturedApp(
-                    name: apps[0].name,
-                    backgroundUrl: apps[0].background,
-                    description: apps[0].description,
-                    onPlay: onPlay
-                  ),
-                  Container(
-                    margin: const EdgeInsets.fromLTRB(0, 60, 0, 0),
-                    child: Row(
-                      mainAxisAlignment: MainAxisAlignment.start,
+            child: FutureBuilder<List<Application>>(
+              future: futureApps,
+              builder: (context, snapshot) {
+                if (snapshot.hasData) {
+                  final apps = snapshot.data;
+                  if(apps!.isEmpty){
+                  return Container(
+                    margin: const EdgeInsets.fromLTRB(0, 250, 0, 0),
+                    alignment: Alignment.center,
+                    child: const Text(
+                       'No app is installed',
+                        textAlign: TextAlign.center,
+                        style: TextStyle(
+                          fontSize: 26,
+                          color: Colors.white,
+                        )
+                      )
+                    );
+                  }
+
+                  return Container(
+                    margin: const EdgeInsets.fromLTRB(0, 30, 0, 0),
+                    constraints: const BoxConstraints(maxWidth: 1100),
+                    alignment: Alignment.center,
+                    child: Column(
                       children: [
+                        FeaturedApp(
+                          id: apps[0].id,
+                          name: apps[0].name,
+                          backgroundUrl: apps[0].background,
+                          description: apps[0].description,
+                          onPlay: widget.onPlay
```

Comment:
You haven't fixed stopping the stream when leaving fullscreen yet, right?

Reply:
Fixed.